nilearn/datasets/neurovault.py

```diff
@@ -0,0 +1,2485 @@
+                          'too many downloads failed in a row ({0})'.format(
+                              n_consecutive_fails))
+            return
+        if found == download_params['max_images']:
+            return
+
+
+# Utilities for providing defaults and transforming input and output
+
+def _split_terms(terms, available_on_server):
+    """Isolate term filters that can be applied by server."""
+    terms_ = dict(terms)
+    server_terms = dict([(k, terms_.pop(k)) for k in
+                         available_on_server if k in terms_ and
+                         (isinstance(terms_[k], _basestring) or
+                          isinstance(terms_[k], int))])
+    return terms_, server_terms
+
+
+def _move_unknown_terms_to_local_filter(terms, local_filter,
+                                        available_on_server):
+    """Move filters handled by the server inside URL.
+
+    Some filters are available on the server and can be inserted into
+    the URL query. The rest will have to be applied on metadata
+    locally.
+
+    """
+    local_terms, server_terms = _split_terms(terms, available_on_server)
+    local_filter = ResultFilter(query_terms=local_terms).AND(local_filter)
+    return server_terms, local_filter
+
+
+def basic_collection_terms():
+    """Return a term filter that excludes empty collections."""
+    return {'number_of_images': NotNull()}
+
+
+def basic_image_terms():
+    """Filter that selects unthresholded F, T and Z maps in mni space
+
+    More precisely, an image is excluded if one of the following is
+    true:
+
+        - It is not in MNI space.
+        - Its metadata field "is_valid" is cleared.
+        - It is thresholded.
+        - Its map type is one of "ROI/mask", "anatomical", or "parcellation".
+        - Its image type is "atlas"
+
+    """
+    return {'not_mni': False, 'is_valid': True, 'is_thresholded': False,
+            'map_type': NotIn('ROI/mask', 'anatomical', 'parcellation'),
+            'image_type': NotEqual('atlas')}
+
+
+def _move_col_id(im_terms, col_terms):
+    """Reposition 'collection_id' term.
+
+    If the collection id was specified in image filters, move it to
+    the collection filters for efficiency.
+
+    This makes specifying the collection id as a keyword argument for
+    ``fetch_neurovault`` efficient.
+
+    """
+    if 'collection_id' not in im_terms:
+        return im_terms, col_terms
+    im_terms = copy(im_terms)
+    col_terms = copy(col_terms)
+    if 'id' not in col_terms:
+        col_terms['id'] = im_terms.pop('collection_id')
+    elif col_terms['id'] == im_terms['collection_id']:
+        col_terms['id'] = im_terms.pop('collection_id')
+    else:
+        warnings.warn('You specified contradictory collection ids, '
+                      'one in the image filters and one in the '
+                      'collection filters')
+    return im_terms, col_terms
+
+
+def _read_download_params(
+    data_dir, download_mode='download_new'
```

Comment:
Looks like some simplification here. I understand that we have lot and lengthy arguments.